quarkus/deployment/src/main/java/org/keycloak/quarkus/deployment/KeycloakProcessor.java

```diff
@@ -305,7 +306,7 @@ private void configureDefaultPersistenceUnitEntities(ParsedPersistenceXmlDescrip
      *
      * @param recorder
      */
-    @Consume(BootstrapConfigSetupCompleteBuildItem.class)
+    @Consume(CryptoProviderInitBuildItem.class)
     @Record(ExecutionTime.STATIC_INIT)
     @BuildStep
     KeycloakSessionFactoryPreInitBuildItem configureProviders(KeycloakRecorder recorder, List<PersistenceXmlDescriptorBuildItem> descriptors) {
```

Comment:
I would avoid creating a built item here and change this method name to `configureKeycloakSessionFactory` to make sure the provider is set before bootstrapping the session factory.

I don't see a reason for creating another built item if its purpose is only for ordering.

Reply:
Yes, I confirm the only point for introducing build item was ordering. I've changed as you suggested. It works as expected and cryptoProvider is initialized before keycloak session factory. Thanks for this tip!